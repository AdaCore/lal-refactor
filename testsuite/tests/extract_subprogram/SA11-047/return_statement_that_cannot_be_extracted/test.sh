# These tests should report no subprogram extraction available
#   Start of IfStmt
extract_subprogram -P default.gpr -S my_package.adb -SL 5 -SC 8 -EL 5 -EC 9 -N Foo
#   Entire IfStmt
extract_subprogram -P default.gpr -S my_package.adb -SL 5 -SC 8 -EL 7 -EC 13 -N Foo
#   IfStmt plus NullStmt
extract_subprogram -P default.gpr -S my_package.adb -SL 5 -SC 8 -EL 8 -EC 11 -N Foo

# These should be successfully extracted
#   Only final ReturnStmt
extract_subprogram -P default.gpr -S my_package.adb -SL 9 -SC 8 -EL 9 -EC 19 -N Foo
#   NullStmt followed by final ReturnStmt
extract_subprogram -P default.gpr -S my_package.adb -SL 8 -SC 8 -EL 9 -EC 19 -N Foo