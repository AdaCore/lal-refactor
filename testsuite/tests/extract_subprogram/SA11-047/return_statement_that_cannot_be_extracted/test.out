Could not extract subprogram my_package.adb : Foo
Could not extract subprogram my_package.adb : Foo
Could not extract subprogram my_package.adb : Foo
Extracting as a ADA_SUBP_KIND_FUNCTION:
my_package.adb
4:1-4:1 
      function Foo return Boolean;

      function Foo return Boolean is
      begin
         return False;
      end Foo;


9:7-9:20 return Foo;

Extracting as a ADA_SUBP_KIND_FUNCTION:
my_package.adb
4:1-4:1 
      function Foo return Boolean;

      function Foo return Boolean is
      begin
         null;
         return False;
      end Foo;


8:7-9:20 return Foo;

